json/src/lib.rs

```diff
@@ -1006,8 +1007,8 @@ pub struct GetAddressInfoResult {
 /// Models the result of "getblockchaininfo"
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct GetBlockchainInfoResult {
-    /// Current network name as defined in BIP70 (main, test, regtest)
-    pub chain: String,
+    /// Current network name as defined in BIP70 (main, test, signet, regtest)
+    pub chain: Network,
```

Comment:
You need to have `#[serde(deserialize_with = "network_from_core_arg")]` and `fn network_from_core_arg` defined for this to work.

Reply:
Ups, my bad, I forgot to put all the pieces together. Should be good now.